Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/BackgroundTask/AddIshBackgroundTask.cs

```diff
@@ -181,48 +218,115 @@ protected override void EndProcessing()
+                            var ishObjects = new XElement("ishobjects");
+                            foreach (var ishObjectsIshRefGroup in ishObjectsGroupedByIshRef)
+                            {
+                                var ishObjectElement = new XElement("ishobject",
+                                    new XAttribute("ishtype", ishObjectsIshRefGroup.First().IshType),
+                                    new XAttribute("ishref", ishObjectsIshRefGroup.First().IshRef));
+                                ishObjects.Add(ishObjectElement);
+
+                                ishObjectsCount++;
+                            }
+
+                            inputData = ishObjects.ToString();
+
+                            break;
+                        }
                 }
-                
-                if (ParameterSetName == "ParameterGroup")
+
+                if (EventDescription.IsNullOrEmpty())
                 {
-                    // Start event 
-                    WriteDebug($"Create StartEvent EventType[{EventType}] EventDescription[{EventDescription}]");
-                    var startEventRequest = new StartEventRequest
-                    {
-                        description = EventDescription,
-                        eventType = EventType,
-                        maximumProgress = _startEventMaxProgress
-                    };
-                    startEventResponse = IshSession.EventMonitor25.StartEvent(startEventRequest);
+                    EventDescription = $"Executing {EventType} for {ishObjectsCount} ishObjects";
                 }
 
-                if (ParameterSetName == "ParameterGroup" && StartAfter.HasValue)
-                { 
+                // Start event 
+                WriteDebug($"Create StartEvent EventType[{EventType}] EventDescription[{EventDescription}]");
+                var startEventRequest = new StartEventRequest
+                {
+                    description = EventDescription,
+                    eventType = EventType,
+                    maximumProgress = _startEventMaxProgress
+                };
+
+                var startEventResponse = IshSession.EventMonitor25.StartEvent(startEventRequest);
+                var backgroundTaskInputData = Encoding.Unicode.GetBytes(ParameterSetName == "IshObjectsGroup" ? inputData : RawInputData);
+
+                if (HashId == null)
+                {
+                    HashId = CalculateHashId(backgroundTaskInputData);
+                }
+
+                if (StartAfter.HasValue)
+                {
                     // Create BackgroundTask
-                    WriteDebug($"Create BackgroundTask EventType[{EventType}] RawInputData.length[{RawInputData.Length}] StartAfter[{StartAfter}]");
+                    WriteDebug($"Create BackgroundTask EventType[{EventType}] RawInputData.length[{inputData}] StartAfter[{StartAfter}]");
```

Comment:
RawInputData.Length[{inputData}
* should be backgroundTaskInputData given line 314
* should be with .Length as we don't want to print 10000 bytes of text here
The before read `WriteDebug($"Create BackgroundTask EventType[{EventType}] RawInputData.length[{RawInputData.Length}] StartAfter[{StartAfter}]");`

Reply:
Resolved with the following
WriteDebug(ParameterSetName == "IshObjectsGroup" ?
    $"Create BackgroundTask EventType[{EventType}] InputData.length[{inputData.Length}] StartAfter[{StartAfter}]" :
    $"Create BackgroundTask EventType[{EventType}] RawInputData.length[{RawInputData.Length}] StartAfter[{StartAfter}]");